src/gui/panels/board_builder_panel.py

```diff
@@ -381,8 +404,23 @@ def _marker_snapshot_list_to_opencv_points(
         return return_value
 
     def _on_build_board_button_click(self, event: wx.CommandEvent) -> None:
-        corners_dict = self.board_builder.build_board()
+        target_board = self.board_builder.build_board(repeatability_testing=False)  # Put as True if data is needed for repeatability testing
         self._render_frame(self.board_builder.detector_poses, self.board_builder.target_poses)
+        target_board_json = {
+            "target_id": target_board.target_id,
+            "markers": [
+                {
+                    "marker_id": marker.marker_id,
+                    "marker_size": self._marker_size,
+                    "points": marker.points
+                } for marker in target_board.markers
+            ]
+        }
```

Comment:
I'm not sure if you're aware of this, but pydantic does support making JSON strings from models. I believe we are on pydantic 1.10, for which case you might want to check the options here: https://docs.pydantic.dev/1.10/usage/exporting_models/